spec/Section 3 -- Type System.md

```diff
@@ -2227,3 +2227,160 @@ to the relevant IETF specification.
 ```graphql example
 scalar UUID @specifiedBy(url: "https://tools.ietf.org/html/rfc4122")
 ```
+
+## Schema Coordinates
+
+SchemaCoordinate :
+
+- TypeCoordinate
+- MemberCoordinate
+- ArgumentCoordinate
+- DirectiveCoordinate
+- DirectiveArgumentCoordinate
+
+TypeCoordinate : Name
+
+MemberCoordinate : Name . Name
+
+ArgumentCoordinate : Name . Name ( Name : )
+
+DirectiveCoordinate : @ Name
+
+DirectiveArgumentCoordinate : @ Name ( Name : )
+
+:: A _schema coordinate_ is a human readable string that uniquely identifies a
+_schema element_ within a GraphQL Schema.
+
+:: A _schema element_ can be a named type, a field, an input field, an enum
+value, a field argument, a directive, or a directive argument defined within a
+schema (including built-in types and directives).
+
+Note: Meta-fields are not defined within a schema, and thus are not _schema
+element_. By extension, an introspection type is not a _schema element_.
+
+:: The _containing element_ of a _schema element_ is the schema element with one
+fewer {Name} token that syntactically contains it. Specifically:
+
+- The containing element of an {ArgumentCoordinate} is a {MemberCoordinate}.
+- The containing element of a {MemberCoordinate} is a {TypeCoordinate}.
+- The containing element of a {DirectiveArgumentCoordinate} is a
+  {DirectiveCoordinate}.
+- {TypeCoordinate} and {DirectiveCoordinate} have no containing element.
+
+A _schema coordinate_ is always unique. Each _schema element_ can be referenced
+by exactly one possible schema coordinate.
+
```

Comment:
```suggestion

Note: To ensure uniqueness, {Ignored} tokens not valid inside a _schema coordinate_.
```

Reply:
@benjie I suppose it would be most correct to add a `lookahead` rule to the grammars above, but that would end up being super super verbose - is there any shorthand for this?

Reply:
In fact:

> Ignored tokens are [...] not referenced in syntactical grammar productions.
> Any amount of Ignored may appear before and after every lexical token

~ https://spec.graphql.org/October2021/#sec-Language.Source-Text.Ignored-Tokens

So we'd want to change this.... but only in the case of when parsing schema coordinates? Which would lead to a tricky change to the parse() function... 🤔 

(almost feels like at this point you'd be easier off with an entirely different and dedicated `parseSchemaCoordinate()` function)

Reply:
One option of course it to fudge it and add a little context to our otherwise(?) context free grammar (lots of prior art in other languages) and add a special rule in the parser that says "if you're about to yield a schema coordinate, bail out if you encountered any {Ignored} tokens"

I'd maybe vote for that?

Reply:
took a stab at implementation here: https://github.com/graphql/graphql-js/pull/4450